CMakeLists.txt

```diff
@@ -635,30 +636,24 @@ if(ENABLE_TESTING)
                      COMMAND
                        ${CMAKE_COMMAND} -E copy_if_different ${CMAKE_CURRENT_BINARY_DIR}/xdgTestHelper${CMAKE_EXECUTABLE_SUFFIX} ${CMAKE_CURRENT_BINARY_DIR}/TestFoundation/xdgTestHelper${CMAKE_EXECUTABLE_SUFFIX}
                      DEPENDS
-                       TestFoundation
                        xdgTestHelper)
-  add_test(NAME
-             TestFoundation
-           COMMAND
-             ${CMAKE_CURRENT_BINARY_DIR}/TestFoundation/TestFoundation
-           WORKING_DIRECTORY
-             ${CMAKE_CURRENT_BINARY_DIR}/TestFoundation)
-  set_tests_properties(TestFoundation
-                       PROPERTIES
-                         ENVIRONMENT
-                           LD_LIBRARY_PATH=${CMAKE_CURRENT_BINARY_DIR}:${FOUNDATION_PATH_TO_XCTEST_BUILD}:${FOUNDATION_PATH_TO_LIBDISPATCH_BUILD}:${FOUNDATION_PATH_TO_LIBDISPATCH_BUILD}/src
-                         DEPENDS
-                         ${CMAKE_CURRENT_BINARY_DIR}/TestFoundation/xdgTestHelper${CMAKE_EXECUTABLE_SUFFIX})
-
   add_custom_command(TARGET TestFoundation
                      POST_BUILD
                      BYPRODUCTS
                        ${CMAKE_CURRENT_BINARY_DIR}/TestFoundation/plutil${CMAKE_EXECUTABLE_SUFFIX}
                      COMMAND
                        ${CMAKE_COMMAND} -E copy_if_different ${CMAKE_CURRENT_BINARY_DIR}/plutil${CMAKE_EXECUTABLE_SUFFIX} ${CMAKE_CURRENT_BINARY_DIR}/TestFoundation/plutil${CMAKE_EXECUTABLE_SUFFIX}
                      DEPENDS
-                       TestFoundation
                        plutil)
+
+  xctest_discover_tests(TestFoundation
+                        COMMAND
+                          ${CMAKE_CURRENT_BINARY_DIR}/TestFoundation/TestFoundation${CMAKE_EXECUTABLE_PREFIX}
```

Comment:
This should be CMAKE_EXECUTABLE_SUFFIX here :p

Reply:
🤦‍♂️ I think I would wait until CMake 3.15 support is merged, so I don't get to get this thinking-with-my-brain thing right.